tst/EnergyPlus/unit/IndoorIceRink.unit.cc

```diff
@@ -0,0 +1,267 @@
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects, false));
+
+    GetIndoorIceRink(*state);
+
+    // For Ice Rink Input:
+    EXPECT_EQ(Rink(NumOfRinks).Name, "INDOOR ICE RINK");
+    EXPECT_EQ(Rink(NumOfRinks).SchedName, "ALWAYSONSCHEDULE");
+    EXPECT_EQ(Rink(NumOfRinks).ZoneName, "MAIN FLOOR");
+    EXPECT_EQ(Rink(NumOfRinks).SurfaceName, "FLOOR");
+    EXPECT_EQ(Rink(NumOfRinks).TubeDiameter, 0.013);
+    EXPECT_EQ(Rink(NumOfRinks).TubeLength, 30);
+    EXPECT_EQ(Rink(NumOfRinks).ControlStrategy, SurfaceTempControl);
+    EXPECT_EQ(Rink(NumOfRinks).hrstofreeze, 3);
+    EXPECT_EQ(Rink(NumOfRinks).deltatemp, 2);
+    EXPECT_EQ(Rink(NumOfRinks).RefrigSetptSched, "REFRIGSCHED");
+    EXPECT_EQ(Rink(NumOfRinks).IceSetptSched, "ICESCHED");
+    EXPECT_EQ(Rink(NumOfRinks).PeopleHeatGainSchedName, "PEOPLEHGSCHED");
+    EXPECT_EQ(Rink(NumOfRinks).PeopleSchedName, "PEOPLESCHED");
+    EXPECT_EQ(Rink(NumOfRinks).MaxNumOfPeople, 100);
+    EXPECT_EQ(Rink(NumOfRinks).LengthRink, 60);
+    EXPECT_EQ(Rink(NumOfRinks).WidthRink, 30);
+    EXPECT_EQ(Rink(NumOfRinks).WaterTemp, 22);
+    EXPECT_EQ(Rink(NumOfRinks).IceThickness, 0.0254);
+    EXPECT_EQ(Rink(NumOfRinks).FloodWaterTemp, 15);
+    EXPECT_EQ(Rink(NumOfRinks).IceSetPointTemp, -3);
+}
+
+//TEST_F(EnergyPlusFixture, IceRink_Resurfacer_GetInput)
+//{
+//    std::string const idf_objects = delimited_string({
+//        "IceRink:Resurfacer,                                                            ",
+//        "Resurfacer,              !- Name                                           ",
+//        "ResurfSched,             !- Resurfacing Schedule Name                      ",
+//        "1,                       !- Number Of Resurfacing Events                   ",
+//        "15,                      !- Resurfacing Water Temperature {C}              ",
+//        "10,                      !- Initial Water Temperature {C}                  ",
+//        "3;                       !- Resurfacer Tank Capacity                       ",
+//    });
+//
+//    ASSERT_TRUE(process_idf(idf_objects, false));
+//
+//    GetResurfacer();
+//
+//    // For Resurfacer Input:
+//    EXPECT_EQ(Resurfacer(NumOfResurfacers).Name, "RESURFACER");
+//    EXPECT_EQ(Resurfacer(NumOfResurfacers).ResurfacingSchedName, "RESURFSCHED");
+//    EXPECT_EQ(Resurfacer(NumOfResurfacers).NoOfResurfEvents, 1);
+//    EXPECT_EQ(Resurfacer(NumOfResurfacers).ResurfacingWaterTemp, 15);
+//    EXPECT_EQ(Resurfacer(NumOfResurfacers).InitWaterTemp, 10);
+//    EXPECT_EQ(Resurfacer(NumOfResurfacers).TankCapacity, 3);
+//}
+
+//TEST_F(EnergyPlusFixture, IceRink_PeopleHG)
+//{
+//    Rink.allocate(1);
+//    Schedule.allocate(2);
+//    Schedule(1).CurrentValue = 1.0;
+//    Schedule(2).CurrentValue = 25;
+//    Rink(1).PeopleHeatGainSchedPtr = 1;
+//    Rink(1).PeopleSchedPtr = 2;
+//    Rink(1).Name = "INDOORICERINK";
+//
+//    Real64 Q_people = Rink(1).PeopleHG();
+//
+//    EXPECT_EQ(Q_people, 25.0);
+//}
```

Comment:
Any reason these had to be commented out?  If they aren't relevant then just delete them.